server/src/main/java/org/elasticsearch/cluster/routing/allocation/DataTier.java

```diff
@@ -188,6 +188,37 @@ public static boolean isFrozenNode(final Set<DiscoveryNodeRole> roles) {
         }
     }
 
+    /**
+     * Compares the provided tiers for coldness order (eg. warm is colder than hot).
+     *
+     * Similar to {@link java.util.Comparator#compare(Object, Object)} returns
+     *   -1 if tier1 is colder than tier2 (ie. compare("data_cold", "data_hot"))
+     *   0 if tier1 is as cold as tier2 (ie. tier1.equals(tier2) )
+     *   1 if tier1 is warmer than tier2 (ie. compare("data_hot", "data_cold"))
+     *
+     * The provided tiers parameters must be valid data tiers values (ie. {@link #ALL_DATA_TIERS}.
+     * NOTE: `data_content` is treated as "equal to data_hot" in the tiers hierarchy.
+     * If invalid tier names are passed the result is non-deterministic.
+     */
+    public static int compare(String tier1, String tier2) {
+        if (tier1.equals(DATA_CONTENT)) {
+            tier1 = DATA_HOT;
+        }
+        if (tier2.equals(DATA_CONTENT)) {
+            tier2 = DATA_HOT;
+        }
```

Comment:
@dakrone handling the case where an index has  the following configuration. Not entirely happy with treating `data_content` as equal to `data_hot` (as they are not comparable, or shouldn't be) but I guess it'll have to do unless you have a better suggestion? 

```
"routing" : {
          "allocation" : {
            "include" : {
              "_tier_preference" : "data_content"
            },
            "require" : {
              "data" : "warm"
            }
          }
        }
```

Reply:
I don't think there's any way around it unfortunately. Functionally though, we consider them to be on the same hardware, so I don't think this is so bad, since it's only for the purpose of (internal) ordering rather than overwriting.